### PR TITLE
Clarify one invalid vs

### PR DESCRIPTION
**Pull request by arianaw66**
An example of how this will look:
Screen Shot 2020-02-25 at 5 47 42 PM

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2228

### CODE REVIEW

File: projects/gloo/cli/pkg/printers/virtualservice.go [hunk 1]
```diff
@@ -130,35 +131,51 @@ func getRouteTableStatus(vs *v1.RouteTable) string {
...
-		return cleanSubResourceError(subResourceErrorMessages[0])
+		subResourceMessage = cleanSubResourceError(subResourceErrorMessages[0])
 	default:
 		// there are multiple errors, don't be fancy, just return list
-		return strings.Join(subResourceErrorMessages, "\n")
+		subResourceMessage = strings.Join(subResourceErrorMessages, "\n")
 	}
+
+	// Note: Parent function does NOT maintain newlines. Keeping them in case we fix that in the future.
+	return resourceStatus.String() + "\n" + subResourceMessage
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/printers/virtualservice.go]: nit, `strings.Join()` a bit more idiomatic here

File: projects/gloo/cli/pkg/printers/virtualservice.go [hunk 2]
```diff
@@ -279,3 +296,7 @@ func genericErrorFormat(resourceName, statusString, reason string) string {
 func subResourceErrorFormat(errorDetails string) string {
 	return fmt.Sprintf("Error with Route: %v: %v", strings.TrimSpace(gloov1.UpstreamListErrorTag), strings.TrimPrefix(errorDetails, ": "))
 }
+func genericSubResourceMessage(resourceName, statusString string) string {
+	return fmt.Sprintf("%v is in a %v state. Updates to this resouce may be blocked by problems with another resource.",
+		resourceName, statusString)
+}
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/printers/virtualservice.go]: we should be specific about when updates may be blocked (by default) or not blocked (with route replacement enabled)

File: projects/gloo/cli/pkg/printers/virtualservice.go (hunk 1 above)
Comment (arianaw66) [reply projects/gloo/cli/pkg/printers/virtualservice.go]: You mean by building a string slice in between? Seems like unnecessary overhead?

Comment (kdorosh) [reply projects/gloo/cli/pkg/printers/virtualservice.go]: true, agreed let's keep it as is

Comment (kdorosh) [reply projects/gloo/cli/pkg/printers/virtualservice.go]: thought there was a version of the function that accepted strings without string slice

**Comment by arianaw66** [comment]
known Kube2e flake 
/kick

**Pull request merged**